jobserver/api/releases.py

```diff
@@ -125,21 +119,28 @@ def get(self, request, workspace_name):
         return Response(generate_index(files))
 
 
-class ReleaseFileAPI(APIView):
+class ReleaseAPI(APIView):
     # DRF file upload does not use multipart, is just a simple byte stream
     parser_classes = [FileUploadParser]
 
-    def put(self, request, release_id, filename):
+    def post(self, request, release_id):
         """Upload a file for this Release.
 
         File must be listed in Release.requested_files, and the hash must match.
         """
         release = get_object_or_404(Release, id=release_id)
+        backend, user = validate_upload_access(request, release.workspace)
 
-        if filename not in release.requested_files:
-            raise NotFound
+        try:
+            upload = request.data["file"]
+            filename = upload.name
```

Comment:
Can this line throw a KeyError? if not can we pull it out of the try/catch to leave it focussed?

Reply:
fixed